deepchem/feat/molecule_featurizers/mat_featurizer.py

```diff
@@ -0,0 +1,92 @@
+from deepchem.feat.base_classes import MolecularFeaturizer
+from deepchem.utils.molecule_feature_utils import one_hot_encode
+try:
+  import logging
+  import numpy as np
+  import pandas as pd
+  import torch
+  from rdkit import Chem
+  from rdkit.Chem import AllChem
+  from rdkit.Chem import MolFromSmiles
+  from sklearn.metrics import pairwise_distances
+  from torch.utils import datasets
+except:
+  raise ImportError("Required Modules not found.")
+
+
+class MATFeaturizer(MolecularFeaturizer):
+  """
+  This class is a featurizer for the Molecule Attention Transformer.
+  """
+  def __init__(
+      self,
+      add_dummy_node=True,
+      one_hot_formal_charge=True,
+  ):
+  """
+  Parameters
+  ----------
+  mol (rdchem.Mol): An RDKit Mol object.
+  add_dummy_node (bool): If True, a dummy node will be added to the molecular graph.
+  one_hot_formal_charge (bool): If True, formal charges on atoms are one-hot encoded.
+  """
+
+    self.mol = mol
+    self.add_dummy_node = add_dummy_node
+    self.one_hot_formal_charge = one_hot_formal_charge
+
+  def atom_features(self, atom, one_hot_formal_charge=True):
+    """
+    Deepchem already contains an atom_features function, however we are defining a new one here due to the need to handle features specific to MAT.
```

Comment:
Can you explain more about what features you are adding that are new? We will also need a parameter docstring added here

Reply:
Done! Please take a look.

Reply:
Ok looks good I think